STUDIO/org.openl.security.standalone/src/org/openl/rules/security/standalone/persistence/User.java

```diff
@@ -125,7 +126,18 @@ public void setPasswordHash(String passwordHash) {
     }
 
     public void setSurname(String surname) {
-        this.surname = surname;
+        this.surname = truncateIfNeeded("surname", surname);
+    }
+
+    private String truncateIfNeeded(String fieldName, String field) {
+        try {
+            int size = getClass().getDeclaredField(fieldName).getAnnotation(Column.class).length();
+            if (field != null && field.length() > size) {
+                return field.substring(0, size);
+            }
+        } catch (NoSuchFieldException e) {
+        }
+        return field;

```

Comment:
Please, keep this logic in the service class.

BTW, you can keep max field lengths in the constants. Please replace reflection with using of constants.

Reply:
Reflection in this simple case looks ugly